.github/workflows/check-pull-request.yml

```diff
@@ -0,0 +1,44 @@
+name: Pull Request - Rust CI with Cargo
+
+on:
+  pull_request:
+    branches:
+      - develop
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Install System Dependencies
+        run: sudo apt-get update && sudo apt-get install -y libasound2-dev         
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+
+      - name: Build using Cargo
+        run: cargo build
```

Comment:
What about using cargo check instead? Just to save time.

> Check a local package and all of its dependencies for errors. This will essentially compile the packages without performing the final step of code generation, which is faster than running cargo build
-- [source](https://doc.rust-lang.org/cargo/commands/cargo-check.html)

Reply:
According the resource you provided me, I agree with you. So, I had changed.